pkg/agent/manager/cache/cache.go

```diff
@@ -117,24 +119,24 @@ func New(log logrus.FieldLogger, trustDomain string, bundle []*x509.Certificate)
 	}
 }
 
-func (c *cacheImpl) SetBundles(newBundles map[string][]*x509.Certificate) {
+func (c *cacheImpl) SetBundles(newBundles map[string]*Bundle) {
 	// SetBundles() and Bundle()/Bundles() can be called concurrently since
 	// the "property" is atomic. Before the following code can merge in changes
 	// it needs to make a copy of the map to mutate so it doesn't modify
 	// the bundle map out from underneath readers. SetBundles() is not intended
 	// to be called by more than one goroutine at a time.
 
 	// copy the map
-	bundles := make(map[string][]*x509.Certificate)
+	bundles := make(map[string]*Bundle)
 	for k, v := range c.Bundles() {
 		bundles[k] = v
 	}
 
 	// merge in changes
 	changed := false
 	for id, newBundle := range newBundles {
-		bundle, ok := bundles[id]
-		if !ok || !certsEqual(bundle, newBundle) {
+		bundle := bundles[id]
+		if bundle == nil || !bundle.EqualTo(newBundle) {
```

Comment:
I guess you're really not a fan of the `ok` pattern :) any particular reason other than small GC optimization? I personally prefer it because I think it's a little clearer, and it consistently applies in all cases (e.g. if the value is not a pointer)

Reply:
ha, depends :) when the thing being stuffed into the map is pointery, i tend to not do comma-ok since the nil test covers two birds with one stone if future refactoring accidentally introduces nil values in the map (although, one could argue panicking under such a case might be ok and help you find the bug introducing nil values sooner).